R/lognormal_ab_test.R

```diff
@@ -1,104 +1,130 @@
-lognormal_ab_test <- function(data, nsim=1e5, alpha0=1, beta0=25, m0=4, k0=1, s_sq0=1, v0=5, expected_revenue_converted_users=NULL, expected_conversion_rate=NULL, plot.density=FALSE, conf.level=.1, tolerance=.01, save.hist=TRUE){  
-  library(ggplot2)
-  #bayes.lognormal.test <- function(data, nsim=1e5, alpha0=1, beta0=1, m0=4, k0=1, s_sq0=1, v0=1, plot.density=FALSE, conf.level=.1, tolerance=.01){  
-  
-  # probability model:  
-  # conversion - binomial likelihood, beta prior with shape parameters alpha0, beta0  
-  # revenue - Normal likelihood model for log(revenue). Prior: 
+lognormal_ab_test <- function(data,
+                              nsim = 1e5,
+                              alpha0 = 1,
+                              beta0 = 25,
+                              m0 = 4, k0 = 1, s_sq0 = 1, v0 = 5,
+                              expected_revenue_converted_users = NULL,
+                              expected_conversion_rate = NULL,
+                              plot.density = FALSE,
+                              conf.level = 0.1,
+                              tolerance = 0.01,
+                              save.hist = TRUE) {
+  # Probability model:
```

Comment:
Sorry the code isn't very transparent. The model is largely based on this
series of blog posts:
http://engineering.richrelevance.com/bayesian-ab-tests/
http://engineering.richrelevance.com/bayesian-analysis-of-normal-distributions-with-python/
http://engineering.richrelevance.com/bayesian-ab-testing-with-a-log-normal-model/

For a deeper mathematical treatment of the Normal model, see chapter 3 of
Gelman's Bayesian Data Analysis.

There are four steps:
1. Model spenders vs nonspenders with a beta-binomial model, like we do for
conversion.
2. Log transform the revenue amongst spenders.
3. Model log(revenue amongst spenders) with a normal model.
4. Transform back to the data space


On Tue, Mar 24, 2015 at 8:47 PM, Christopher Peters <
notifications@github.com> wrote:

> In R/lognormal_ab_test.R
> <https://github.com/uken/testr/pull/3#discussion_r27087415>:
>
> > -
> > -  # probability model:
> > -  # conversion - binomial likelihood, beta prior with shape parameters alpha0, beta0
> > -  # revenue - Normal likelihood model for log(revenue). Prior:
> > +lognormal_ab_test <- function(data,
> > +                              nsim = 1e5,
> > +                              alpha0 = 1,
> > +                              beta0 = 25,
> > +                              m0 = 4, k0 = 1, s_sq0 = 1, v0 = 5,
> > +                              expected_revenue_converted_users = NULL,
> > +                              expected_conversion_rate = NULL,
> > +                              plot.density = FALSE,
> > +                              conf.level = 0.1,
> > +                              tolerance = 0.01,
> > +                              save.hist = TRUE) {
> > +  # Probability model:
>
> This is totally not finished yet. @ayakubovich
> <https://github.com/ayakubovich> to be candid, I need to go through this
> and teach myself what you're doing here. I can see that you're using this
> to test for differences in revenue, but at the moment it's just code to me.
> Do you have any sources handy?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/uken/testr/pull/3/files#r27087415>.
>